src/sentry/api/endpoints/client_state.py

```diff
@@ -0,0 +1,55 @@
+# This endpoint helps managing persisted client state with a TTL for a member, organization or user
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.utils import json, redis
+
+STATE_CATEGORIES = {
+    "onboarding": {
+        "ttl": 30 * 24 * 60 * 60,  # the time in seconds that the state will be persisted
+        "scope": "member",  # Can be "user", "org", or "user-org"
+    }
+}
+
+
+class ClientStateEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated,)
+
+    def get_key(self, request: Request, category):
+        scope = STATE_CATEGORIES[category]["scope"]
+        org = request.query_params.get("org")
+        if scope == "member":
+            if not org:
+                raise ParameterValidationError("Missing org parameter")
+            return f"client-state:{category}:{org}:{request.user.id}"
+        elif scope == "org":
+            if not org:
+                raise ParameterValidationError("Missing org parameter")
+            return f"client-state:{category}:{org}"
+        else:
+            return f"client-state:{category}:{request.user.id}"
+
+    def get(self, request: Request, category) -> Response:
+        if category not in STATE_CATEGORIES:
+            return Response({"detail": "Category not found"}, status=404)
+        key = self.get_key(request, category)
+        res = None
+        with redis.clusters.get("default").map() as client:
+            res = client.get(key)
```

Comment:
@Neo-Zhixing what happens if there is a key error?

Reply:
it won't because we do validate that `category` is in the list

Reply:
that said i can probably move the validation logic to within `get_key`

Reply:
@Neo-Zhixing well you might have a valid category but what if there is no value yet inserted for that key? As in, you access the value before we write it to the cache.

Reply:
Then `client.get(key)` would just return a Promise that resolves to None, which i check down below with `if res.value:`, in which case we return `{}`
And Redis's `expire` is a noop for keys that don't exist.

Reply:
@Neo-Zhixing got it, let's make sure we type the front end to include `None` as a possible return type for this API always.